File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String kNumber = transformToKNumber(n, k);
        String[] split = kNumber.split("0");
        for (String str : split) {
            if (!str.equals("") && isPrimeNumber(str)) {
                answer += 1;
            }
        }

        return answer;
    }

    private String transformToKNumber(int n, int k) {
        int current = n;
        String answer = "";
        while (current > 0) {
            answer = current % k + answer;
            current = current / k;
        }

        return answer;
    }

    private boolean isPrimeNumber(String str) {
        long num = Long.parseLong(str);
        if (num == 2 || num == 3) {
            return true;
        } else if (num % 2 == 0 || num < 2) {
            return false;
        }

        for (int i = 3; i < (int) Math.sqrt(num) + 1; i++) {
            if (num % i == 0) return false;
        }

        return true;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution(0, 3));
    }
}

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def to_k_number(n, k):
    ret = ""
    while n > 0:
        ret += str(n % k)
        n //= k
    return ''.join(reversed(ret))

def is_prime_number(k):
    if k == 2 or k == 3:
        return True
    if k % 2 == 0 or k < 2:
        return False
    
    for i in range(3, int(k ** 0.5) + 1, 2):
        if k % i == 0:
            return False
    return True
    
def solution(n, k):
    answer = 0
    k_num = to_k_number(n, k)
    
    for num in k_num.split('0'):
        if num == "":
            continue
        if is_prime_number(int(num)):
            answer += 1
            
    
    return answer